neqo-transport/src/connection/mod.rs

```diff
@@ -3124,7 +3128,8 @@ impl Connection {
         self.tps.borrow().remote.as_ref().map_or_else(
             || Duration::new(0, 0),
             |r| {
-                let exponent = u32::try_from(r.get_integer(tparams::ACK_DELAY_EXPONENT)).unwrap();
+                let exponent = u32::try_from(r.get_integer(tparams::ACK_DELAY_EXPONENT))
+                    .expect("ACK_DELAY_EXPONENT > 20 is invalid per RFC9000");
                 Duration::from_micros(v.checked_shl(exponent).unwrap_or(u64::MAX))
```

Comment:
Note that this `checked_shl` is not right.  The checking only ensures that the shift value is within range (i.e., between 0 and 64).  It doesn't catch overflow.  Given that `v` could be up to 2<sup>62</sup>-1, there's a real overflow risk.

Perhaps the right answer here is:

```rust
let exponent = u32::try_from(r.get_integer(tparams::ACK_DELAY_EXPONENT)).expect("your message");
let corrected = if v.leading_zeros() >= exponent {
  v << exponent
} else {
  u64::MAX
};
Duration::from_micros(corrected)
```